crates/bevy_transform/src/systems.rs

```diff
@@ -339,33 +339,32 @@ mod test {
     #[test]
     #[should_panic]
     fn panic_when_hierarchy_cycle() {
-        let mut app = App::new();
+        let mut world = World::default();
+        let mut update_stage = SystemStage::single_threaded();
```

Comment:
```suggestion
        // This test is run on a single thread in order to avoid breaking the global task pool by panicking
        // This fixes the flaky tests reported in https://github.com/bevyengine/bevy/issues/4996
        let mut update_stage = SystemStage::single_threaded();
```